spec/sidekiq-rate-limiter/fetch_patch_spec.rb

```diff
@@ -1,59 +1,35 @@
 require 'spec_helper'
 require 'sidekiq'
 require 'sidekiq/api'
+require 'support/job'
+require 'support/proc_job'
+require 'sidekiq-rate-limiter/fetch_patch'
 
-RSpec.describe Sidekiq::RateLimiter::Fetch do
-  before(:all) do
-    class Job
-      include Sidekiq::Worker
-      sidekiq_options 'queue'    => 'basic',
-                      'retry'    => false,
-                      'rate' => {
-                          'limit'  => 1,
-                          'period' => 1
-                      }
-      def perform(*args); end
-    end
-    class ProcJob
-      include Sidekiq::Worker
-      sidekiq_options 'queue'    => 'basic',
-                      'retry'    => false,
-                      'rate' => {
-                          'limit'  => ->(arg1, arg2) { arg2 },
-                          'name'   => ->(arg1, arg2) { arg2 },
-                          'period' => ->(arg1, arg2) { arg2 }
-                      }
-      def perform(arg1, arg2); end
-    end
-  end
-
+RSpec.describe Sidekiq::FetchPatch do
   let(:options)       { { queues: [queue, another_queue, another_queue] } }
   let(:queue)         { 'basic' }
   let(:another_queue) { 'some_other_queue' }
   let(:args)          { ['I am some args'] }
   let(:worker)        { Job }
   let(:proc_worker)   { ProcJob }
   let(:redis_class)   { Sidekiq.redis { |conn| conn.class } }
-
-  it 'should inherit from Sidekiq::BasicFetch' do
-    expect(described_class).to be < Sidekiq::BasicFetch
-  end
+  let(:patched_class) { Sidekiq::BasicFetch.class_eval { prepend Sidekiq::FetchPatch } }
```

Comment:
Hm - this `let()` has global side effects.

Could we subclass `Sidekiq::BasicFetch` instead so it's scoped to the individual test?

```
let(:patched_class) { Class.new(Sidekiq::BasicFetch) { include Sidekiq::FetchPatch } }
```

Reply:
Nice pick up, this is actually important. The other spec `server_spec.rb` is actually failed, because we haven't fire the :startup event, so the patch was not load. But if we run  `fetch_patch_spec.rb` before `server_spec.rb`, it patched the Sidekiq::BasicFetch already, then the all specs would pass.